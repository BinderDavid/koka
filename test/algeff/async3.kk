// --------------------------------------------------------
// Async demo: do concurrently
// --------------------------------------------------------

function test() {
  println("what is your name?")
  val xs = asyncAll( [
    { print("what is your name?")
      readline()
    },
    { print("what is your age?")
      readline()
    }
  ])
  println("hi " + xs.join(","))  
}

function main() {
  async(test)
}