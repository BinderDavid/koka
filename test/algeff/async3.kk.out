main : () -> <asyn,console> ()
test : () -> <asyn,console> ()